src/hooks/amm.ts

```diff
@@ -0,0 +1,93 @@
+import { Provider } from "@ethersproject/providers";
+import { Token } from "@token";
+import { BigNumberish, ethers, Signer } from "ethers";
+import { useSignerFromEthereum } from "./signer";
+import SWAP_ABI from "@abis/swap.json";
+import {
+  getTimeMinutesFromNow,
+  uiToNative,
+} from "./utils";
+
+/**
+ * @notice Call the `swap` function of the `Swap` contract
+ * @param poolName the name of the pool to use
+ * @param chainId the id of the chain to swap on
+ * @param signer the signer who is calling the swap
+ * @param tokenFrom the token to swap from
+ * @param tokenTo the token to swap to
+ * @param amountFrom the amount of tokenFrom to swap
+ * @param amountTo optional: the minimum amount of tokenTo to receive
+ * @param deadline optional: the deadline for the txn to execute
+ * @returns amount of tokenTo received from the swap
+ */
+async function swap(
+  poolName: any,
+  chainId: number,
+  signer: Signer,
+  tokenFrom: Token,
+  tokenTo: Token,
+  amountFrom: number, // in units if ui
+  amountTo?: number, // in units of ui
+  deadline?: number
+) {
+  // Get the swap contract
+  const { swapAddress, swapEthAddress } = usePoolTokenInfo(poolName, chainId);
```

Comment:
currently `usePoolTokenInfo()` is not a correct way to get the swapAddress. waiting for the PR from metagabbar with the tokens and chains to add

Reply:
Isn't `hooks` react specific ? I thought we wanted this SDK to cater to general JS bridging functionality.

Reply:
resolving: i think this is the right place to keep it? i don't think hooks are only react specific, but might be wrong